optuna/artifact/exceptions.py

```diff
@@ -0,0 +1,9 @@
+class ArtifactNotFound(Exception):
```

Comment:
[nit] I'm wondering if the `ArtifactNotFound` exception should inherit the `OptunaError` exception. Any opinions on this?

Reply:
As `StorageInternalError` inherits the `OptunaError`, it seems more natural to inherit `OptunaError`.